lib/model/autocomplete.dart

```diff
@@ -274,58 +245,134 @@ class MentionAutocompleteView extends ChangeNotifier {
           isDone = true;
           break;
         }
-
-        final User user = iterator.current;
-        if (query.testUser(user, store.autocompleteViewManager.autocompleteDataCache)) {
-          results.add(UserMentionAutocompleteResult(userId: user.userId));
-        }
+        final T item = iterator.current;
+        final result = testItem(query, item);
+        if (result != null) results.add(result);
       }
     }
     return results;
   }
 }
 
-class MentionAutocompleteQuery {
-  MentionAutocompleteQuery(this.raw, {this.silent = false})
-    : _lowercaseWords = raw.toLowerCase().split(' ');
+class MentionAutocompleteView extends AutocompleteView<MentionAutocompleteQuery, MentionAutocompleteResult, User> {
```

Comment:
The data type here will be changed eventually when adding other ***mentionable types*** but I think leaving it as `User` for now is okay.